9 спринт

IvanPodlesnykh opened the pull request.

IvanPodlesnykh committed:
Приложение Playlist Maker
1. Убрал .idea из git

IvanPodlesnykh committed:
Приложение Playlist Maker
Первое ДЗ, добавил темную тему для главного экрана и экрана настроек, перенес атрибуты в стили.

IvanPodlesnykh committed:
Приложение Playlist Maker
Второе и третье ДЗ:
1. Добавил действие кнопке "Пользовательское соглашение" на экране настроек
2. Добавил действие кнопке "Написать в поддержку" на экране настроек
3. Реализовал вёрстку экрану поиска

IvanPodlesnykh committed:
Приложение Playlist Maker
Четвертое ДЗ:
1. Добавил действие кнопке "Поделиться приложением" на экране настроек
2. Добавил кнопку очистки поискового запроса
3. Добавил сохранение и восстановление поискового запроса

nasibullin1412 commented:
⚠️Кнопки на экране настроек имеют слишком большой отступ относительно правого края

![image](https://github.com/IvanPodlesnykh/PlaylistMaker/assets/57040048/fa660ad1-3d44-4d73-b900-0adb17480b7e)

nasibullin1412 commented on app/src/main/java/com/ivanpodlesnykh/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,78 @@
 
 class SearchActivity : AppCompatActivity() {
+
+    private var searchString: String = SEARCH_STRING_DEF
```
🍏Лучше такие поля делать нулабельными и инициализировать `null`, так как `null` в синтаксисе языка Kotlin проще отличить от пустой строки

nasibullin1412 commented on app/src/main/java/com/ivanpodlesnykh/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,78 @@
+    companion object {
+        const val SEARCH_STRING_KEY = "SEARCH_INPUT"
+        const val SEARCH_STRING_DEF = ""
+    }
```
🍏`companion object` лучше объявлять в конце класса. Также поля не используются вне класса `SearchActivity`, значит их можно сделать `private`

nasibullin1412 commented on app/src/main/java/com/ivanpodlesnykh/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,78 @@
+        const val SEARCH_STRING_DEF = ""
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
```
🍏Старайся не засорять метод `onCreate`, лучше раскидать всё по отдельным методам, которые будут вызываться отсюда

nasibullin1412 commented on app/src/main/java/com/ivanpodlesnykh/playlistmaker/SettingsActivity.kt:
```diff
@@ -16,5 +18,28 @@ class SettingsActivity : AppCompatActivity() {
+            val intent = Intent(Intent.ACTION_SENDTO, Uri.parse("mailto:"))
+            intent.putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.dev_mail)))
+            intent.putExtra(Intent.EXTRA_SUBJECT, getString(R.string.mail_to_dev_topic))
+            intent.putExtra(Intent.EXTRA_TEXT, getString(R.string.mail_to_dev_text))
```
🍏Когда идёт работа с одной и той же переменной или полем подряд старайся использовать `apply` или `with`, например это можно переписать на
```
            val intent = Intent(Intent.ACTION_SENDTO, Uri.parse("mailto:")).apply {
                putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.dev_mail)))
                putExtra(Intent.EXTRA_SUBJECT, getString(R.string.mail_to_dev_topic))
                putExtra(Intent.EXTRA_TEXT, getString(R.string.mail_to_dev_text))
            }
```

nasibullin1412 commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,7 +1,62 @@
+        android:orientation="horizontal">
+        <ImageView
+            android:id="@+id/search_back_button"
+            android:layout_width="48dp"
+            android:layout_height="48dp"
```
🍏 48dp лучше вынести в ресурсы

IvanPodlesnykh committed:
Приложение Playlist Maker
Правки 9 спринт:
1. Реализовал переключатель темной темы
2. Изменил отступы от правого края для кнопок настроек
3. Обернул обработку кнопок в функции и вынес их из функции onCreate

IvanPodlesnykh committed:
Приложение Playlist Maker
Правки 9 спринт:
1. Убрал эффект нажатия на кнопки на экране настроек
